src/G2/Language/AST.hs

```diff
@@ -309,8 +309,8 @@ instance ASTContainer DataCon Expr where
     modifyContainedASTs _ d = d
 
 instance ASTContainer DataCon Type where
-    containedASTs (DataCon _ t) = [t]
-    modifyContainedASTs f (DataCon n t) = DataCon n (f t)
+    containedASTs (DataCon _ t u e) = [t] ++ concatMap containedASTs u ++ concatMap containedASTs e
+    modifyContainedASTs f (DataCon n t u e) = DataCon n (f t) (map (modifyContainedASTs f) u) (map (modifyContainedASTs f) e)
```

Comment:
Don't need to use concatMap/map- just call containedASTs/modifyContainedASTs on the whole lists